x-pack/packages/kbn-elastic-assistant/impl/data_anonymization_editor/context_editor_modal/index.tsx

```diff
@@ -0,0 +1,242 @@
+import { SelectedPromptContext } from '../../assistant/prompt_context/types';
+import { BatchUpdateListItem } from '../context_editor/types';
+import { updateSelectedPromptContext, getIsDataAnonymizable } from '../helpers';
+import { useAssistantContext } from '../../assistant_context';
+import { bulkUpdateAnonymizationFields } from '../../assistant/api/anonymization_fields/bulk_update_anonymization_fields';
+import { useFetchAnonymizationFields } from '../../assistant/api/anonymization_fields/use_fetch_anonymization_fields';
+
+export interface Props {
+  onClose: () => void;
+  onSave: (updates: BatchUpdateListItem[]) => void;
+  promptContext: SelectedPromptContext;
+}
+
+const SelectedPromptContextEditorModalComponent = ({ onClose, onSave, promptContext }: Props) => {
+  const { http, toasts } = useAssistantContext();
+  const [checked, setChecked] = useState(false);
+  const checkboxId = useGeneratedHtmlId({ prefix: 'updateSettingPresetsCheckbox' });
+
+  const { data: anonymizationFields, refetch: anonymizationFieldsRefetch } =
+    useFetchAnonymizationFields();
+  const [contextUpdates, setContextUpdates] = React.useState<BatchUpdateListItem[]>([]);
+  const [selectedPromptContext, setSelectedPromptContext] = React.useState(promptContext);
+  const [anonymizationFieldsBulkActions, setAnonymizationFieldsBulkActions] =
+    useState<PerformBulkActionRequestBody>({
+      create: [],
+      update: [],
+      delete: {},
+    });
+
+  const isDataAnonymizable = useMemo<boolean>(
+    () => getIsDataAnonymizable(selectedPromptContext.rawData),
+    [selectedPromptContext.rawData]
+  );
+
+  const handleSave = useCallback(async () => {
+    if (onSave) {
+      onSave(contextUpdates);
+    }
+    try {
+      await bulkUpdateAnonymizationFields(http, anonymizationFieldsBulkActions, toasts);
+      anonymizationFieldsRefetch();
+    } catch (e) {
+      /* empty */
+    }
+    onClose();
+  }, [
+    anonymizationFieldsBulkActions,
+    anonymizationFieldsRefetch,
+    contextUpdates,
+    http,
+    onClose,
+    onSave,
+    toasts,
+  ]);
+
+  const onListUpdated = useCallback(
+    (updates: BatchUpdateListItem[]) => {
+      setContextUpdates((prev) => [...prev, ...updates]);
+
+      setAnonymizationFieldsBulkActions((prev) => {
+        return updates.reduce<PerformBulkActionRequestBody>(
+          (acc, item) => {
+            const persistedField = find(anonymizationFields.data, ['field', item.field]) as
+              | AnonymizationFieldResponse
+              | undefined;
+
+            if (persistedField) {
+              acc.update?.push({
+                id: persistedField.id,
+                ...(item.update === 'allow' || item.update === 'defaultAllow'
+                  ? { allowed: item.operation === 'add' }
+                  : {}),
+                ...(item.update === 'allowReplacement' || item.update === 'defaultAllowReplacement'
```

Comment:
Hmm... I thought we got rid of this terminology in favor of `allow` and `anonymize` https://github.com/elastic/kibana/pull/179701

Reply:
This appears to be the same block that @YulNaumenko used over in https://github.com/elastic/kibana/pull/179701, and this was pushed on APR-12 after Yuliia's changes were merged on APR-11, so I imagine this is as-intended. I will double check and test though.

https://github.com/elastic/kibana/blob/97448bd65a6553df54e15998bed4ebd43b303516/x-pack/packages/kbn-elastic-assistant/impl/data_anonymization/settings/anonymization_settings/index.tsx#L41-L49